src/ast/codegen_llvm.cpp

```diff
@@ -875,6 +883,25 @@ void CodegenLLVM::visit(FieldAccess &acc)
   }
 }
 
+void CodegenLLVM::visit(ArrayIndex &arr)
+{
+  Value *array, *index, *offset;
+  SizedType &type = arr.expr->type;
+
+  arr.expr->accept(*this);
+  array = expr_;
+
+  arr.indexpr->accept(*this);
+  index = b_.CreateIntCast(expr_, b_.getInt64Ty(), false); // promote int to 64-bit
+  offset = b_.CreateMul(index, b_.getInt64(type.pointee_size));
+
+  AllocaInst *dst = b_.CreateAllocaBPF(SizedType(Type::integer, type.pointee_size), "array_index");
```

Comment:
```suggestion
  AllocaInst *dst = b_.CreateAllocaBPF(SizedType(Type::integer, type.pointee_size), "array_access");
```

Reply:
thanks for calling this out 100% missed it on my first rename pass.